dex/src/main/scala/com/wavesplatform/dex/api/MatcherWebSocketRoute.scala

```diff
@@ -74,21 +78,23 @@ case class MatcherWebSocketRoute(addressDirectory: ActorRef, matcher: ActorRef,
       }
       .watchTermination()(handleTermination)
 
-  private def signedGet(prefix: String, publicKey: PublicKey): Directive0 = parameters(('Timestamp, 'Signature)).tflatMap {
-    case (timestamp, signature) =>
-      Base58
-        .tryDecodeWithLimit(signature)
-        .map { crypto.verify(_, prefix.getBytes(StandardCharsets.UTF_8) ++ publicKey.arr ++ Longs.toByteArray(timestamp.toLong), publicKey) } match {
-        case Success(true) => pass
-        case _             => complete(InvalidSignature)
-      }
-  }
+  private def signedGet(prefix: String, publicKey: PublicKey, incorrectSignatureAction: Directive0 = complete(InvalidSignature)): Directive0 =
+    parameters(('t, 's)).tflatMap {
+      case (timestamp, signature) =>
+        Base58
+          .tryDecodeWithLimit(signature)
+          .map { crypto.verify(_, prefix.getBytes(StandardCharsets.UTF_8) ++ publicKey.arr ++ Longs.toByteArray(timestamp.toLong), publicKey) } match {
+          case Success(true) => pass
+          case _             => incorrectSignatureAction
+        }
+    }
 
   /** Requires PublicKey, Timestamp and Signature of [prefix `as`, PublicKey, Timestamp] */
   private def accountUpdates: Route = (path("accountUpdates" / PublicKeyPM) & get) { publicKey =>
-    signedGet("as", publicKey) {
-      handleWebSocketMessages(Flow.fromSinkAndSourceCoupled(Sink.ignore, accountUpdatesSource(publicKey)))
+    val directive = optionalHeaderValueByName(`X-Api-Key`.name).flatMap { maybeKey =>
```

Comment:
Well, the right code here is:
```scala
withAuth | signedGet(balanceStreamPrefix, publicKey) // note, it won't work with in inverse order
```
It reflects the task description: if we have an api key, use it and ignore a signature. Otherwise check the request by a signature.

Reply:
Done